### DISCUSSION TITLE
Firmware with webrepl/wifi enabled

### DISCUSSION THREAD

**Discussion by Skorpeo** [micropython/micropython#12533] [ESP32]
Hi, I am flashing a esp32 device over the air and was wondering how I can enable webrepl by default in the firmware.  I don't want to open the device and deal with cables, its also not so simple, so I was hoping there is a way or perhaps instructions to build firmware with webrepl/wifi enabled automatically?

**Comment by robert-hh** [comment]
You can place the respective code for starting up the internet and webrepl into the file modules/_boot.py, which is frozen into the firmware when you built it. _boot.py is executed when the device boots. A password can be supplied with webrepl.start(password="<your password>"). But then it cannot be changed, If you need to change it, you have to adapt webrepl.py, e.g. to use the password supplied in the call to webrepl,start() as default, if webrepl_cfg.py is not present.
